pyartifactory/objects.py

```diff
@@ -1,952 +1,961 @@
+        """
+        Lists all the groups
+        :return: GroupList
+        """
+        response = self._get(f"api/{self._uri}")
+        logger.debug("List all groups successful")
+        return [Group(**group) for group in response.json()]
+
+    def update(self, group: Group) -> Group:
+        """
+        Updates an exiting group in Artifactory with the provided group details.
+        :param group: Group to be updated
+        :return: Updated group
+        """
+        group_name = group.name
+        self.get(group_name)
+        self._post(f"api/{self._uri}/{group_name}", json=group.dict())
+        logger.debug("Group %s successfully updated", group_name)
+        return self.get(group_name)
+
+    def delete(self, name: str) -> None:
+        """
+        Removes a group
+        :param name: Name of the group to delete
+        :return: None
+        """
+        self.get(name)
+        self._delete(f"api/{self._uri}/{name}")
+        logger.debug("Group %s successfully deleted", name)
+
+
+class ArtifactoryRepository(ArtifactoryObject):
+    """Models an artifactory repository."""
+
+    _uri = "repositories"
+
+    # Repositories operations
+    def get_repo(self, repo_name: str) -> AnyRepositoryResponse:
+        """
+        Finds repository in artifactory. Raises an exception if the repo doesn't exist.
+        :param repo_name: Name of the repository to retrieve
+        :return: Either a local, virtual or remote repository
+        """
+        try:
+            response = self._get(f"api/{self._uri}/{repo_name}")
+            repo: AnyRepositoryResponse = parse_obj_as(
+                Union[
+                    LocalRepositoryResponse,
+                    VirtualRepositoryResponse,
+                    RemoteRepositoryResponse,
+                ],
+                response.json(),
+            )
+            return repo
+        except requests.exceptions.HTTPError as error:
+            if error.response.status_code == 404 or error.response.status_code == 400:
+                logger.error("Repository %s does not exist", repo_name)
+                raise RepositoryNotFoundException(
+                    f" Repository {repo_name} does not exist"
+                )
+            raise ArtifactoryException from error
+
+    def create_repo(self, repo: AnyRepository) -> AnyRepositoryResponse:
+        """
+        Creates a local, virtual or remote repository
+        :param repo: Either a local, virtual or remote repository
+        :return: LocalRepositoryResponse, VirtualRepositoryResponse or RemoteRepositoryResponse object
+        """
+        repo_name = repo.key
+        try:
+            self.get_repo(repo_name)
+            logger.error("Repository %s already exists", repo_name)
+            raise RepositoryAlreadyExistsException(
+                f"Repository {repo_name} already exists"
+            )
+        except RepositoryNotFoundException:
+            data= json.dumps(repo, default=custom_encoder)

```

Comment:
The pre-commits also highlight formatting issue in this line. Indeed `data=` should be `data =`